### PR TITLE
Add support for using application credentials for authentication

### PR DESCRIPTION
**Pull request by scornelissen85**
I've used this code for using application credentials for authentication in Openstack. I think it would be valuable if this could be added in the fog-openstack project.

This will fix issue #510

This can be used as alternative next to it's original credentials options. To use application credentials I've set the following parameters:

```
compute_client = Fog::Compute.new  :openstack_auth_url => ENV['OS_AUTH_UR'],
                                    :openstack_application_credential_secret => ENV['OS_APPLICATION_CREDENTIAL_SECRET'],
                                    :openstack_application_credential_id => ENV['OS_APPLICATION_CREDENTIAL_ID']
```

CC @dhague (sorry if I should cc another member)

### CODE REVIEW

File: lib/fog/openstack/auth/token/v3.rb [hunk 1]
```diff
@@ -50,6 +55,7 @@ def path
           end
 
           def scope
+            return false if @application_credential
```
Comment (geemus) [inline lib/fog/openstack/auth/token/v3.rb:58]: It feels to me like `nil` might be a bit more appropriate than false here. It should still return falsey in the conditional, but just conceptually a nil scope makes more sense to me than a false scope. Does that make sense?

**Comment by geemus** [comment]
Looks good overall.

Unfortunately the test suite has been busted for a bit here, other maintainers haven't really been active, and I don't have an active openstack setup to test against either.

Have you been able to test this out successly?

**Comment by scornelissen85** [comment]
> Looks good overall.
> 
> Unfortunately the test suite has been busted for a bit here, other maintainers haven't really been active, and I don't have an active openstack setup to test against either.
> 
> Have you been able to test this out successly?

Yes, I'm using the branch with these changes with success and only use application credentials to log in in Openstack. I'm using this branch currently from beaker-openstack where the Compute and the Network class is used directly. I'm not sure if something is missing as I'm not using anything other than those classes.

Furthermore, I'm not an expert about the communication with Openstack. This change works for me, but I can't guarantee it works for other cases.

File: lib/fog/openstack/auth/token/v3.rb (hunk 1 above)
Comment (scornelissen85) [reply lib/fog/openstack/auth/token/v3.rb:58]: Looks fine to me to change it to `nil`. I will test if it then still works.

**Comment by geemus** [comment]
Thanks for the context, that makes sense to me. The change to nil would be great and then I think we can bring this in. You can only speak for your own experience, but from what I see it shouldn't break anything that is currently working for others either way.

**Comment by scornelissen85** [comment]
I have changes `scope` to return `nil` when using application credentials and it seems to work.

**Comment by geemus** [comment]
Great, thanks for that.

**Review by geemus — APPROVED** [review]

**Comment by geemus** [comment]
@scornelissen85 did you need a release with this change?

**Comment by scornelissen85** [comment]
@geemus If you could make a release I can then make a PR for beaker-openstack to use the application credentials.

**Pull request merged**